src/mapping.rs

```diff
@@ -31,6 +31,24 @@ impl<'s> ParseError<'s> {
     }
 }
 
+impl fmt::Display for ParseError<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.kind {
```

Comment:
I'm wondering if we should also format the line, maybe with `f.alternate()` set? Like this:
```
invalid utf8: blabla in "<line>"
```

Reply:
hm, not sure how useful that is. I think in practice, this error will most likely occur when you just *try* to parse *any* kind of file, to determine if its even valid proguard, like the `is_valid` fn does. In that case you will just have random binary garbage.